Library/ViewModels/ManageViewPledgeViewModel.swift

```diff
@@ -38,13 +48,30 @@ public final class ManageViewPledgeViewModel:
 
     self.configureRewardSummaryView = projectAndReward
       .map(second)
+
+    let project = projectAndReward.map(first)
+
+    self.showActionSheetMenuWithOptions = project
+      .takeWhen(self.menuButtonTappedSignal)
+      .map { project -> [ActionSheetMenuOption] in
```

Comment:
This almost makes me think this should be an extension (maybe `private` extension on `Project`)...something along these lines..

```swift
private extension Project {
  func managePledgeOptions() -> [ActionSheetMenuOption] {
    switch self.state {
    case: .live: .allCases
    default: [.contactCreator]
    }
  }
}
```

Then we'd be able to do

```swift
.map(project.managePledgeOptions) // not sure if it does point free - wasn't using compiler when writing this
```

Dunno which one is better (this is pretty easy to read so I'd be fine with it).

Reply:
I'm not sure I agree with this - the list of alert options is very view-specific, I would be hesitant to couple it so directly with `Project` model. I'm going to leave it in the view model if that's ok.